### PR TITLE
Added AccTests for MicrosoftAD to StorageGateway

### PR DESCRIPTION
**Pull request by AdamTylerLynch**
### Community Note

* Please vote on this pull request by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original pull request comment to help the community and maintainers prioritize this request
* Please do not leave "+1" or other comments that do not add relevant new information or questions, they generate extra noise for pull request followers and do not help prioritize the request

Added acceptance tests to StorageGateway Gateway for Managed Microsoft Active Directory.

Output from acceptance testing:

```
$ make testacc TESTARGS='-run=TestAccAWSStorageGatewayGateway_SmbMicrosoft'
==> Checking that code complies with gofmt requirements...
TF_ACC=1 go test ./aws -v -count 1 -parallel 20 -run=TestAccAWSStorageGatewayGateway_SmbMicrosoft -timeout 180m
=== RUN   TestAccAWSStorageGatewayGateway_SmbMicrosoftActiveDirectorySettings
=== PAUSE TestAccAWSStorageGatewayGateway_SmbMicrosoftActiveDirectorySettings
=== RUN   TestAccAWSStorageGatewayGateway_SmbMicrosoftActiveDirectorySettings_timeout
=== PAUSE TestAccAWSStorageGatewayGateway_SmbMicrosoftActiveDirectorySettings_timeout
=== CONT  TestAccAWSStorageGatewayGateway_SmbMicrosoftActiveDirectorySettings
=== CONT  TestAccAWSStorageGatewayGateway_SmbMicrosoftActiveDirectorySettings_timeout
--- PASS: TestAccAWSStorageGatewayGateway_SmbMicrosoftActiveDirectorySettings (2149.32s)
--- PASS: TestAccAWSStorageGatewayGateway_SmbMicrosoftActiveDirectorySettings_timeout (2199.40s)
PASS
ok  	github.com/terraform-providers/terraform-provider-aws/aws	2201.254s

...
```

### CODE REVIEW

**Comment by AdamTylerLynch** [comment]
Fixed the structure of the AccTests for DRY.

make testacc TESTARGS='-run=TestAccAWSStorageGatewayGateway_Smb -parallel=10'  
==> Checking that code complies with gofmt requirements...
TF_ACC=1 go test ./aws -v -count 1 -parallel 20 -run=TestAccAWSStorageGatewayGateway_Smb -parallel=10 -timeout 180m
=== RUN   TestAccAWSStorageGatewayGateway_SmbActiveDirectorySettings
=== PAUSE TestAccAWSStorageGatewayGateway_SmbActiveDirectorySettings
=== RUN   TestAccAWSStorageGatewayGateway_SmbActiveDirectorySettings_timeout
=== PAUSE TestAccAWSStorageGatewayGateway_SmbActiveDirectorySettings_timeout
=== RUN   TestAccAWSStorageGatewayGateway_SmbMicrosoftActiveDirectorySettings
=== PAUSE TestAccAWSStorageGatewayGateway_SmbMicrosoftActiveDirectorySettings
=== RUN   TestAccAWSStorageGatewayGateway_SmbMicrosoftActiveDirectorySettings_timeout
=== PAUSE TestAccAWSStorageGatewayGateway_SmbMicrosoftActiveDirectorySettings_timeout
=== RUN   TestAccAWSStorageGatewayGateway_SmbGuestPassword
=== PAUSE TestAccAWSStorageGatewayGateway_SmbGuestPassword
=== CONT  TestAccAWSStorageGatewayGateway_SmbActiveDirectorySettings
=== CONT  TestAccAWSStorageGatewayGateway_SmbMicrosoftActiveDirectorySettings_timeout
=== CONT  TestAccAWSStorageGatewayGateway_SmbActiveDirectorySettings_timeout
=== CONT  TestAccAWSStorageGatewayGateway_SmbGuestPassword
=== CONT  TestAccAWSStorageGatewayGateway_SmbMicrosoftActiveDirectorySettings
--- PASS: TestAccAWSStorageGatewayGateway_SmbGuestPassword (334.16s)
--- PASS: TestAccAWSStorageGatewayGateway_SmbActiveDirectorySettings (1017.23s)
--- PASS: TestAccAWSStorageGatewayGateway_SmbActiveDirectorySettings_timeout (1080.99s)
--- PASS: TestAccAWSStorageGatewayGateway_SmbMicrosoftActiveDirectorySettings_timeout (2178.46s)
PASS
ok  	github.com/terraform-providers/terraform-provider-aws/aws	2234.050s

File: aws/resource_aws_storagegateway_gateway_test.go [hunk 1]
```diff
@@ -462,6 +462,67 @@ func TestAccAWSStorageGatewayGateway_SmbActiveDirectorySettings_timeout(t *testi
...
+	resourceName := "aws_storagegateway_gateway.test"
+
+	resource.ParallelTest(t, resource.TestCase{
+		PreCheck:     func() { testAccPreCheck(t) },
+		ErrorCheck:   testAccErrorCheck(t, storagegateway.EndpointsID),
+		Providers:    testAccProviders,
+		CheckDestroy: testAccCheckAWSStorageGatewayGatewayDestroy,
+		Steps: []resource.TestStep{
+			{
+				Config: testAccAWSStorageGatewayGatewayConfig_SmbMicrosoftActiveDirectorySettings(rName),
```
Comment (DrFaust92) [inline aws/resource_aws_storagegateway_gateway_test.go:477]: can we add an update check as well? for both tests if possible

**Review by DrFaust92 — APPROVED** [review]
LGTM

```
--- PASS: TestAccAWSStorageGatewayGateway_SmbMicrosoftActiveDirectorySettings (2247.93s)
--- PASS: TestAccAWSStorageGatewayGateway_SmbMicrosoftActiveDirectorySettings_timeout (2370.27s)
```

File: aws/resource_aws_storagegateway_gateway_test.go (hunk 1 above)
Comment (AdamTylerLynch) [reply aws/resource_aws_storagegateway_gateway_test.go:477]: When you say an update check, can you please clarify what you would like to see? Feel free to point me at another AccTest for a different resource if that's easiest.

**Review by ewbankkit — APPROVED** [review]
LGTM :rocket:.

##### Commercial

```console
% make testacc TEST=./aws TESTARGS='-run=TestAccAWSStorageGatewayGateway_SmbMicrosoft'      
==> Checking that code complies with gofmt requirements...
TF_ACC=1 go test ./aws -v -count 1 -parallel 20 -run=TestAccAWSStorageGatewayGateway_SmbMicrosoft -timeout 180m
=== RUN   TestAccAWSStorageGatewayGateway_SmbMicrosoftActiveDirectorySettings
=== PAUSE TestAccAWSStorageGatewayGateway_SmbMicrosoftActiveDirectorySettings
=== RUN   TestAccAWSStorageGatewayGateway_SmbMicrosoftActiveDirectorySettings_timeout
=== PAUSE TestAccAWSStorageGatewayGateway_SmbMicrosoftActiveDirectorySettings_timeout
=== CONT  TestAccAWSStorageGatewayGateway_SmbMicrosoftActiveDirectorySettings
=== CONT  TestAccAWSStorageGatewayGateway_SmbMicrosoftActiveDirectorySettings_timeout
--- PASS: TestAccAWSStorageGatewayGateway_SmbMicrosoftActiveDirectorySettings (1142.79s)
--- PASS: TestAccAWSStorageGatewayGateway_SmbMicrosoftActiveDirectorySettings_timeout (1198.23s)
PASS
ok  	github.com/terraform-providers/terraform-provider-aws/aws	1203.228s
```

##### GovCloud

```console
% make testacc TEST=./aws TESTARGS='-run=TestAccAWSStorageGatewayGateway_SmbMicrosoft'
==> Checking that code complies with gofmt requirements...
TF_ACC=1 go test ./aws -v -count 1 -parallel 20 -run=TestAccAWSStorageGatewayGateway_SmbMicrosoft -timeout 180m
=== RUN   TestAccAWSStorageGatewayGateway_SmbMicrosoftActiveDirectorySettings
=== PAUSE TestAccAWSStorageGatewayGateway_SmbMicrosoftActiveDirectorySettings
=== RUN   TestAccAWSStorageGatewayGateway_SmbMicrosoftActiveDirectorySettings_timeout
=== PAUSE TestAccAWSStorageGatewayGateway_SmbMicrosoftActiveDirectorySettings_timeout
=== CONT  TestAccAWSStorageGatewayGateway_SmbMicrosoftActiveDirectorySettings
=== CONT  TestAccAWSStorageGatewayGateway_SmbMicrosoftActiveDirectorySettings_timeout
--- PASS: TestAccAWSStorageGatewayGateway_SmbMicrosoftActiveDirectorySettings_timeout (1077.82s)
--- PASS: TestAccAWSStorageGatewayGateway_SmbMicrosoftActiveDirectorySettings (1098.68s)
PASS
ok  	github.com/terraform-providers/terraform-provider-aws/aws	1101.862s
```

**Pull request merged**